### PR TITLE
add cd_files to vsphere builder

### LINKED ISSUE #8778: Add support for creating a cd iso version of floppy_files to vmware-iso builder
## Feature Description

Not every version of linux compiles the floppy kernel modules with the live cd installers. I would be nice to have a version of `floppy_files` but builds a iso and mounts that as an additional disc drive in addition to the iso.

#### Use Case(s)

See issue https://github.com/clearlinux/distribution/issues/1769

I am trying to setup ClearLinux with packer in a CI environment hosted in docker, so I do not have access to an external ip/port that I can use with `http_files`. Additionally I am using the `vmware-iso` builder which lets me add `iso_paths` but that iso must already live on the remote esxi datastore. I would need the ability to upload that iso file via packer if I were to make it myself and want the files to live inside the IaC git repo.

It seems like it would be alot more steps to figure out how to upload a temp iso file, mount it via `iso_paths` and then delete it after the VM is finished setting up.

Thanks for reading.

### PR DESCRIPTION
**Pull request by SwampDragons**
add cd_files to vsphere builder.

Closes #8778

### CODE REVIEW

File: builder/vsphere/iso/step_add_cdrom.go [hunk 1]
```diff
@@ -62,6 +62,14 @@ func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multiste
 		}
 	}
 
+	// Add our custom CD, if it exists
+	cd_path, ok := state.Get("cd_path").(string)
+	if ok {
+		if cd_path != "" {
+			s.Config.ISOPaths = append(s.Config.ISOPaths, cd_path)
+		}
+	}
```
Comment (azr) [inline builder/vsphere/iso/step_add_cdrom.go:71]: NIT, we can ignore wether the type assertion was succesful and here cd_path will always be a string :
```suggestion
	// Add our custom CD, if it exists
	if cd_path, _ := state.Get("cd_path").(string) ; cd_path != "" {
		s.Config.ISOPaths = append(s.Config.ISOPaths, cd_path)
	}
```

**Review by azr — APPROVED** [review]
LGTM !

File: builder/vsphere/iso/step_add_cdrom.go (hunk 1 above)
Comment (SwampDragons) [reply builder/vsphere/iso/step_add_cdrom.go:71]: Not true because CD path doesn't always exist, so it could be a nil interface.

Comment (azr) [reply builder/vsphere/iso/step_add_cdrom.go:71]: @SwampDragons do you have a playground example where that would be the case ? In my last example I tried to cover all cases but may be I missed something 👀 .

Comment (SwampDragons) [reply builder/vsphere/iso/step_add_cdrom.go:71]: 🤦‍♀️ no I just can't read. You're right.

**Pull request merged**